Sources/Core/AWSClientRuntime/Auth/CredentialsProviding.swift

```diff
@@ -8,16 +8,18 @@
 import ClientRuntime
 
 /// A type that can provide credentials for authenticating with an AWS service
-public protocol CredentialsProviding {
-    /// Returns credentials for authenticating with an AWS service
-    func getCredentials() async throws -> Credentials
```

Comment:
Swift is in pre-GA status which allows breaking changes and Amplify, the main customer atm, understands this as well so I think it's better to clean up old API now instead of making the old API wrap the new one.